yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -819,12 +819,22 @@ private[spark] class Client(
       javaOpts += "-XX:CMSIncrementalDutyCycle=10"
     }
 
-    // Include driver-specific java options if we are launching a driver
-    if (isClusterMode) {
```

Comment:
In the cluster mode, the memory settings provided for the driver are applied. And in the client mode, the driver and am settings are applied individually. So in both cases we have to check the driver settings.

We haven't moved the (isClusterMode) check, its introduced to ensure that the supplied extraJavaOptions are added when running in the cluster mode only and are not included in the client mode.